transformers/modeling_bert.py

```diff
@@ -840,20 +840,20 @@ def forward(self, input_ids, attention_mask=None, token_type_ids=None, position_
         #    of predictions for masked words.
         # 2. If encoder hidden states are provided we are in a causal situation where we
         #    try to predict the next word for each input in the encoder.
-        if masked_lm_labels is not None and encoder_hidden_states is not None:
+        if masked_lm_labels is not None and lm_labels is not None:
```

Comment:
No, please allow both. No need to restrict, free the researchers!

Reply:
Just remove this test and raised error.

Reply:
free the researchers!